### ISSUE TITLE
Issue with IP configuration using purefa_network and netaddr compatibility

### ISSUE THREAD

**Issue by ryuyamag** [Pure-Storage-Ansible/FlashArray-Collection#629]
I am currently attempting to configure IP addresses using the purefa_network playbook, but I have encountered an issue related to the version of the netaddr module. Specifically, when using netaddr versions 1.2.0 or 1.3.0, errors occur during execution, whereas version 1.1.0 does not cause any problems.

Ansible version : core 2.15.8
FlashArray Collection version: 1.31.1
Purity version: 6.6.7
FlashArray module causing the failure: purefa_network
An example task that fails: set or remove ipaddr/mask/gateway 
Response from ansible-playbook -vvvv when running a failing task

```
fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.229.192.60'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.229.192.60'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.229.192.60'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.229.192.60'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.229.192.60'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.229.192.60'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.229.192.60'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.229.192.60'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.229.192.60'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings\n  warnings.warn(\nTraceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1725709231.220546-56982-269451338359497/AnsiballZ_purefa_network.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1725709231.220546-56982-269451338359497/AnsiballZ_purefa_network.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1725709231.220546-56982-269451338359497/AnsiballZ_purefa_network.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.purestorage.flasharray.plugins.modules.purefa_network', init_globals=dict(_module_fqn='ansible_collections.purestorage.flasharray.plugins.modules.purefa_network', _modlib_path=modlib_path),\n  File \"/usr/lib64/python3.9/runpy.py\", line 225, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib64/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_purestorage.flasharray.purefa_network_payload_e1og9so3/ansible_purestorage.flasharray.purefa_network_payload.zip/ansible_collections/purestorage/flasharray/plugins/modules/purefa_network.py\", line 790, in <module>\n  File \"/tmp/ansible_purestorage.flasharray.purefa_network_payload_e1og9so3/ansible_purestorage.flasharray.purefa_network_payload.zip/ansible_collections/purestorage/flasharray/plugins/modules/purefa_network.py\", line 760, in main\n  File \"/tmp/ansible_purestorage.flasharray.purefa_network_payload_e1og9so3/ansible_purestorage.flasharray.purefa_network_payload.zip/ansible_collections/purestorage/flasharray/plugins/modules/purefa_network.py\", line 352, in update_interface\n  File \"/usr/local/lib/python3.9/site-packages/netaddr/strategy/ipv4.py\", line 99, in valid_str\n    raise TypeError('Invalid type: %s' % type(addr))\nTypeError: Invalid type: <class 'NoneType'>\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

**Comment by sdodsley** [comment]
This appears related to a fix applied to `netaddr` 1.2.0.
We will implement a fix for the module.

**Comment by ryuyamag** [comment]
I confirmed that the issue has been resolved.
Thank you for creating the fix.
